src/components/node-list/components/row/row.js

```diff
@@ -0,0 +1,113 @@
+import React from 'react';
+import { uniqueId } from 'lodash';
+import classnames from 'classnames';
+import NodeIcon from '../../../icons/node-icon';
+import VisibleIcon from '../../../icons/visible';
+import InvisibleIcon from '../../../icons/invisible';
+import FocusModeIcon from '../../../icons/focus-mode';
+import { ToggleControl } from '../../../ui/toggle-control/toggle-control';
+import { RowText } from '../../../ui/row-text/row-text';
+
+import './row.scss';
+
+const Row = ({
+  active,
+  checked,
+  children,
+  dataTest,
+  disabled,
+  faded,
+  focused,
+  focusModeIcon = FocusModeIcon,
+  highlight,
+  icon,
+  id,
+  invisibleIcon = InvisibleIcon,
+  isSlicingPipelineApplied,
+  kind,
+  label,
+  name,
+  onChange,
+  onClick,
+  onMouseEnter,
+  onMouseLeave,
+  onToggleHoveredFocusMode,
+  parentClassName,
+  rowType,
+  selected,
+  type,
+  visibleIcon = VisibleIcon,
+}) => {
+  const isModularPipeline = type === 'modularPipeline';
+  const FocusIcon = isModularPipeline ? focusModeIcon : null;
+  const isChecked = isModularPipeline ? checked || focused : checked;
+  const VisibilityIcon = isChecked ? visibleIcon : invisibleIcon;
+
+  return (
+    <form
```

Comment:
hey the original intention is because the `ToggleControl` is the `label` and `input`, that's why the parent will need to be wrapped in `form` element. Do you think that shouldn't be?

Reply:
Okay got it. I missed the `ToggleControl` as child for <form>.